### PR TITLE
KEP-272 Implemented Value size limit

### PR DESCRIPTION
**Pull request by rnistuk**

### CODE REVIEW

File: raft/test/raft_test.cpp [hunk 1]
```diff
@@ -21,11 +21,11 @@ using namespace ::testing;
 
 namespace
 {
-    const bzn::uuid_t TEST_NODE_UUID{"f0645cc2-476b-485d-b589-217be3ca87d5"};
+    const bzn::uuid_t k_TEST_NODE_UUID{"f0645cc2-476b-485d-b589-217be3ca87d5"};
```
Comment (ebruck) [inline raft/test/raft_test.cpp]: Not sure we need the k_ as all caps usually indicates const and our compiler will tell us so if we try to modify it.

File: storage/storage.cpp [hunk 2]
```diff
@@ -50,6 +50,11 @@ storage::error_msg(storage_base::result error_id)
         case storage_base::result::exists:
             message = "exists";
             break;
+        case storage_base::result::value_too_large:
+            message = "value exceeds ";
```
Comment (ebruck) [inline storage/storage.cpp]: No need to break this into three lines:

mesage = "value exceeds " + std::to_string(MAX_VALUE_SIZE) + "  kb" 

should work fine.

If you stick with this then please put the code in a braces such as:

case ASDF:
{
    ...
    break;
}

File: storage/storage.hpp [hunk 3]
```diff
@@ -40,7 +40,7 @@ namespace bzn
 
         storage_base::result load(const std::string& path) override;
 
-        std::string error_msg(storage_base::result error_id) override;
+        static std::string error_msg(storage_base::result error_id);// override;
```
Comment (ebruck) [inline storage/storage.hpp]: I would not pollute the public interface with this function. I'd put it into the unnamed namespace in the cpp file instead.
There is no need to make this a static and public.

File: storage/storage_base.hpp [hunk 4]
```diff
@@ -64,7 +66,7 @@ namespace bzn
 
         virtual storage_base::result load(const std::string& path) = 0;
 
-        virtual std::string error_msg(storage_base::result error_id) = 0;
+        //virtual std::string error_msg(storage_base::result error_id) = 0;
```
Comment (ebruck) [inline storage/storage_base.hpp]: remove dead code

File: crud/test/crud_tests.cpp [hunk 5]
```diff
@@ -126,7 +126,7 @@ class crud_test : public Test
 
 TEST_F(crud_test, test_that_follower_not_knowing_leader_fails_to_create)
 {
-    auto request = generate_create_request(user_uuid, "key0", k_TEST_VALUE);
+    auto request = generate_create_request(k_USER_UUID, "key0", k_TEST_VALUE);
```
Comment (ebruck) [inline crud/test/crud_tests.cpp]: not a fan of k_

File: raft/test/raft_test.cpp (hunk 1 above)
Comment (rnistuk) [reply raft/test/raft_test.cpp]: I will remove the k_ 's

File: storage/storage.cpp (hunk 2 above)
Comment (rnistuk) [reply storage/storage.cpp]: I have combined the code into one line.

File: crud/test/crud_tests.cpp (hunk 5 above)
Comment (rnistuk) [reply crud/test/crud_tests.cpp]: Removed the k_ wart.

File: storage/storage.hpp (hunk 3 above)
Comment (rnistuk) [reply storage/storage.hpp]: removed the error_msg function.

File: storage/test/storage_test.cpp [hunk 6]
```diff
@@ -21,14 +21,15 @@
...
 
 using namespace ::testing;
 
 namespace
 {
-    const bzn::uuid_t user_uuid = "4bba2aeb-44fe-441e-bb6b-8817561eb716";
-    const std::string key = "bluzelle.json";
-    const std::string value = "ewogICJsaXN0ZW5lcl9hZGRyZXNzIiA6ICIxMjcuMC4wLjEiLAogICJsaXN0ZW5lcl9wb3J0IiA6"
+    const bzn::uuid_t USER_UUID = "4bba2aeb-44fe-441e-bb6b-8817561eb716";
+    const std::string k_KEY = "bluzelle.json";
```
Comment (ebruck) [inline storage/test/storage_test.cpp]: missed one

**Pull request merged**